### ISSUE TITLE
Strange issue: Two URLs being downloaded at the same time

### ISSUE THREAD

**Issue by UnforeseenOcean** [RipMeApp/ripme#26]
* Ripme version: 1.5.2
* Java version (output of `java -version`): 1.7.0_79
* Operating system: Windows 7 SP1
* Exact URL you were trying to rip when the problem occurred: Multiple (`oneballharry.tumblr.com` and `bearlounge.tumblr.com`)
* Please include any additional information about how to reproduce the problem:
Try to download multiple URLs, including those two links on top of each other

## Expected Behavior
It downloads one link, Stop button is functional, then moves to next link

## Actual Behavior
It tries to download two URLs simultaneously and Stop button is disabled (as shown by disk monitor, log and pending counter jumping between two ranges)

**Comment by rautamiekka** [comment]
Check your download thread setting in RipMe.

And **_why_** are you using **_an ancient_** Java ?

**Comment by dikshaa42** [comment]
how did you try this? We can not add multiple url, it will say wrong format. Please tell us how did you do that?

**Comment by UnforeseenOcean** [comment]
Answers to above two questions:

1. *Check your download thread setting in RipMe.*

Yep, but as far as I can understand it's only meant to accelerate **THIS** download only, not start something else before current one finishes. It's set to 5.


2. *how did you try this? We can not add multiple url, it will say wrong format. Please tell us how did you do that?*

Type 1 URL, then press Rip. Then put another link, then press Rip button again. It adds them to the queue.

**Comment by rautamiekka** [comment]
^ I'm at this very moment using RipMe to fetch my favs folders from DeviantArt, which means half a dozen links, no problem. The difference is I'm using Java SE 8 Update 141.

**Comment by UnforeseenOcean** [comment]
First of all,
![error](https://user-images.githubusercontent.com/11834016/28717851-62a8c846-73de-11e7-9b12-df497eeef18f.png)

And second, I will try again after updating.

**Comment by rautamiekka** [comment]
^ That pic makes no sense.

**Comment by UnforeseenOcean** [comment]
The picture is saying "Oh. My brain crashed."

And after updating, I don't think I can replicate the error.
But here's the thing; Could any of you code-savvy people check the code to see if it does handle queues as they are supposed to?

They're supposed to finish one download THEN move to the next link, not download one more simultaneously.

**Comment by rautamiekka** [comment]
^ If I understand you right, it's ripping simultaneously from `oneballharry.tumblr.com` **and** `bearlounge.tumblr.com` **instead of** getting **multiple files at once** from `oneballharry.tumblr.com` **before** `bearlounge.tumblr.com` ?

**Comment by UnforeseenOcean** [comment]
Yes. Exactly.
![image 093](https://user-images.githubusercontent.com/11834016/28724530-459b64a4-73f5-11e7-9454-dc84ecfcb465.png)

**Comment by dikshaa42** [comment]
<img width="534" alt="screen shot 2017-07-29 at 7 29 14 pm" src="https://user-images.githubusercontent.com/12117644/28745337-4e775f2a-7494-11e7-9812-fbc4f9b0fef7.png">

It is working fine .

**Comment by UnforeseenOcean** [comment]
Maybe it's something related to OS interaction and stuff? I really have no idea why or how is this happening.

**Comment by metaprime** [comment]
Can you confirm, is the problem only happening when you clicked stop and then tried to resume? I notice strange behavior with that sometimes. The handling of queues / stop / resume might have some bugs. The common use case for the program is to download one thing at a time. FWIW if your downloaded images are getting sent to the right folders, it's not really wrong, except that you're potentially downloading files you don't actually want.

The workaround for the latter scenario is to close the app, restart it, and rip only the things you want.

@rautamiekka I've noticed you're on a crusade against older Java versions. Definitely admirable to get folks updated. We ask what version not so we can evangelize a crusade to get people to update but because it's _possible_ older Java can have bugs that affect correctness in RipMe, but it's not substantially likely in general. Consider this: we don't operate on the fringes of the language or standard library, which means the Java language and JVM features we are using are well-tested and widely-used. Updates are not **_as_** likely to fix bugs in those areas as they are to fix security bugs around the edges or add entirely new language features. Since most web browsers entirely block Java apps from running and RipMe is not a server app, for our purposes, security issues are not a major concern, and new language features are not either because AFAIK we still stick to 1.6-compatible code. That said, 1.7 is quite old temporally and semantically, and is certainly old enough (as it is the previous semantic minor revision), that bug fixes may have occurred in areas we care about. **However, we can be nicer about how we invite people to update their version of Java.**

**Comment by metaprime** [comment]
@rautamiekka: I've noticed you're on a crusade against older Java versions. **Definitely admirable to get folks updated.** A few things to note: we ask what version people are running, not so we can evangelize a crusade to get people to update, but because it's _possible_ older Java can have bugs that affect correctness in RipMe, but that's not substantially likely in general, in my experience. Consider this: we don't operate on the fringes of the language or standard library, which means the Java language and JVM features we are using are well-tested and widely-used. Updates are not **_as_** likely to fix bugs in those areas as they are to fix security bugs around the edges (e.g. preventing arbitrary malicious untrusted Java code from running in Java applets in the browser) or add entirely new language features. These days, most web browsers entirely block Java apps from running when downloaded from the web into the browser, and RipMe is not a server app, so for our purposes, security issues are not a major concern, and new language features are not either because AFAIK we still stick to 1.6-compatible code. That said, 1.7 is quite old temporally and semantically, and is certainly old enough (as it is the previous semantic minor revision), that bug fixes may have occurred in areas we care about. **However, we can be nicer about how we invite people to update their version of Java.** For example, rather than asking "why?" [1], I'd take an approach like this:

> The version of Java you're running (1.7.0_79) is from 2015-04-14, and that's old enough that it might be the cause of a bug. Please update to the latest version of Java ([Java 1.8 Update 144](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)) and confirm that this bug still occurs for you.

(Dates from https://en.wikipedia.org/wiki/Java_version_history#Java_SE_7)

[1] Besides, the most likely answer to "why?": When your web browser doesn't allow you to run Java, Java doesn't tell you to update anymore (seems like a recent thing, maybe it's because I uncheck the box to ask me to update or install via Ninite which does the same), and you've got nothing running Java, there's no reason to update, and most people (even developers of non-Java software) would not think to update on their own.

**Comment by UnforeseenOcean** [comment]
@metaprime 
No, it happens when normally using without stopping. The rip button does work, but the Stop button doesn't work when that happens.
The queue SHOULD be processed one by one but it somehow managed to download two or more things at the same time. It might have something to do with thread management.

**Comment by rautamiekka** [comment]
@metaprime RipMe may not but Java is used for other things too and ppl tend to totally forget Java could be updated and then are surprised why nothing works -or- refuse to update when the app complains a much newer Java is needed, thinking they already have the newest.

**Comment by metaprime** [comment]
@rautamiekka Trying a basic repro on your end with your latest Java should be able to tell you whether a problem exists. If you want to go any further than that, you can ask the person reporting the issue to update their Java and try again.

**Comment by metaprime** [comment]
@UnforeseenOcean I'm not sure it's a bug per se for the queue to be processing two downloads simultaneously. In fact, considering all of the anti-rate-limiting delays we have added, it probably makes more sense to do multiple downloads in parallel (if connecting to different domains). I can see why it would be annoying that the logs get conflated, but your downloaded files should all be going to the correct directories, right?

**Comment by soloturn** [comment]
outdated
